app/jobs/shipit/continuous_delivery_job.rb

```diff
@@ -7,6 +7,9 @@ class ContinuousDeliveryJob < BackgroundJob
     on_duplicate :drop
 
     def perform(stack)
+      # We reload to ensure that, if the record is locked to create a deploy, the following short ciruits work correctly
+      stack = stack.reload
```

Comment:
I am not sure exactly, but assuming that `stack` is a object from a GlobalID, it should be fresh out of the database at this point, right? Why is the reload needed? Local caching?

Reply:
It's [this](https://github.com/Shopify/shipit-engine/pull/1042#issuecomment-608118022) issue (sorry, should have linked). tl;dr the hydrated model instance here could be stale, because of the work of `BackgroundJob::Unique`.

Reply:
Actually, now I think about it, because we don't specify a timeout on the job I don't _think_ that applies.

https://github.com/Shopify/shipit-engine/blob/da691856a1033c2b7cd107cd640c50d7ffdc7ae0/app/jobs/shipit/background_job/unique.rb#L16-L26

Without a timeout `Redis::Lock::LockTimeout` is raised immediately I think 🤔

Reply:
[Yes it is](https://github.com/nateware/redis-objects/blob/1b1d289fdc1dd4585f52fd8f3e6fcf6f1026f40a/lib/redis/lock.rb#L76-L87), so I'll take the reload out.